modules/freewheelSSPBidAdapter.js

```diff
@@ -0,0 +1,314 @@
+  // force ad parsing on the given vast xml
+  '    vastLoader.parseAds(vast, {' +
+  '      onSuccess: function() {' + script + ' }' +
+  '    });' +
+  '  };' +
+  '  document.head.appendChild(st);' +
+  '})();' +
+  '</script>';
+}
+
+var getInBannerScript = function(bid, size) {
+  return 'var config = {' +
+  '      preloadedVast:vast,' +
+  '      autoPlay:true' +
+  '    };' +
+  '    var ad = new window.com.stickyadstv.vpaid.Ad(document.getElementById("freewheelssp_prebid_target"),config);' +
+  '    (new window.com.stickyadstv.tools.ASLoader(' + bid.params.zoneId + ', \'' + getComponentId(bid.params.format) + '\')).registerEvents(ad);' +
+  '    ad.initAd(' + size[0] + ',' + size[1] + ',"",0,"","");';
+};
+
+var getOutstreamScript = function(bid) {
+  var placementCode = bid.adUnitCode;
+
+  var config = bid.params;
+
+  // default placement if no placement is set
+  if (!config.hasOwnProperty('domId') && !config.hasOwnProperty('auto') && !config.hasOwnProperty('p') && !config.hasOwnProperty('article')) {
+    config.domId = placementCode;
+  }
+
+  var script = 'var config = {' +
+  '  preloadedVast:vast,' +
+  '  ASLoader:new window.com.stickyadstv.tools.ASLoader(' + bid.params.zoneId + ', \'' + getComponentId(bid.params.format) + '\')';
+
+  for (var key in config) {
+    // dont' send format parameter
+    // neither zone nor vastUrlParams value as Vast is already loaded
+    if (config.hasOwnProperty(key) && key !== 'format' && key !== 'zone' && key !== 'zoneId' && key !== 'vastUrlParams') {
+      script += ',' + key + ':"' + config[key] + '"';
+    }
+  }
+  script += '};' +
+
+  'window.com.stickyadstv.' + getAPIName(bid.params.format) + '.start(config);';
+
+  return script;
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['video'],
+  aliases: ['stickyadstv'], //  former name for freewheel-ssp
+  /**
+  * Determines whether or not the given bid request is valid.
+  *
+  * @param {object} bid The bid to validate.
+  * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.zoneId);
+  },
+
+  /**
+  * Make a server request from the list of BidRequests.
+  *
+  * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+  * @return ServerRequest Info describing the request to the server.
+  */
+  buildRequests: function(bidRequests) {
+    // var currency = config.getConfig(currency);
+
+    var currentBidRequest = bidRequests[0];
+    if (bidRequests.length > 1) {
+      utils.logMessage('Prebid.JS - freewheel bid adapter: only one ad unit is required.');
+    }
+
+    var requestParams = {
+      reqType: 'AdsSetup',
+      protocolVersion: '2.0',
+      zoneId: currentBidRequest.params.zoneId,
+      componentId: getComponentId(currentBidRequest.params.format)
+    };
+
+    var location = utils.getTopWindowUrl();
+    if (isValidUrl(location)) {
```

Comment:
Why are you validating that `location.href` is a valid URL?  That seems unnecessary....

Reply:
We don't want to receive invalid urls. If loaded inside some iframe we could retreive location : "about:blank" for instance.